### ISSUE TITLE
simplifiedAutoLink breaks when the link contains underscores

### ISSUE THREAD

**Issue by vitosamson** [showdownjs/showdown#384]
Simplified auto links are broken when the link contains underscores.

In the latest version published on npm (1.6.4), the underscores get converted to `em` tags both inside the link and the link text:

```
> new showdown.Converter({ simplifiedAutoLink: true }).makeHtml('http://google.com/link_foo_bar')
'<p><a href="http://google.com/link<em>foo</em>bar">http://google.com/link<em>foo</em>bar</a></p>'
```

Using latest master from this repo, no link is created and the underscores get converted to `em` tags:

```
> new showdown.Converter({ simplifiedAutoLink: true }).makeHtml('http://google.com/link_foo_bar')
'<p>http://google.com/link<em>foo</em>bar</p>'
```

**Comment by tivie** [comment]
## TL;DR;

Hey @vitosamson. It's not broken. To disable underscores being interpreted as emphasis you should enable the option `literalMidWordUnderscores`


## further explanations

Underscores are markdown _magic_ characters and delimit emphasis. Mid word emphasis is allowed in markdown. empahasis and strong take precedence over auto links.

The option `literalMidWordUnderscores` disables this behavior, so that autolinking takes precedence over emphasis and strong.

**Comment by tivie** [comment]
http://showdownjs.github.io/demo/#/http%3A%2F%2Fgoogle.com%2Flink_foo_bar

**Comment by vitosamson** [comment]
Thanks, I missed that option in the docs.

What about the second repro case I provided? With `simplifiedAutoLinks` set to `true` and `literalMidWordUnderscores` set to `false`, no link is created at all.
